src/opnsense/mvc/app/views/OPNsense/Unbound/stats.volt

```diff
@@ -0,0 +1,165 @@
+                td.innerHTML = data[descKey];
+
+                tr.append(th);
+                tr.append(td);
+                parent.append(tr);
+            } else {
+                writeDescs(parent, data[descKey], descriptions[descKey]);
+            }
+        });
+    }
+
+    /**
+     * fetch Unbound stats
+     */
+    function updateStats() {
+        ajaxGet("/api/unbound/diagnostics/stats",
+            {}, function (data, status) {
+                if (status === "success") {
+                    // Clear old view
+                    let statsView = $("#statsView");
+                    statsView.html('');
+
+                    // Sort the keys in order to ensure that Thread 0 will come before Thread 1.
+                    let dataKeys = Object.keys(data['data']);
+                    dataKeys.sort();
+                    dataKeys.forEach(function(key) {
+                        let value = data['data'][key];
+                        if (key === 'total' || key.substr(0, 6) === 'thread') {
+                            let description;
+                            if (key === 'total') {
+                                description = 'Total';
+                            } else {
+                                description = 'Thread ' + key.substr(6);
+                            }
+
+                            let title = document.createElement("h2");
+                            title.innerHTML = description;
+                            statsView.append(title);
+
+                            let table = document.createElement('table');
+                            table.classList.add('table');
+                            table.classList.add('table-striped');
+                            table.style.width = 'auto';
+                            let tbody = document.createElement('tbody');
+                            writeDescs(tbody, value, descriptionMapThread);
+                            table.append(tbody);
+                            statsView.append(table);
+                        } else if (key === "time") {
+                            let title = document.createElement("h2");
+                            title.innerHTML = "Times";
+                            statsView.append(title);
+
+                            let table = document.createElement('table');
+                            table.classList.add('table');
+                            table.classList.add('table-striped');
+                            table.style.width = 'auto';
+                            let tbody = document.createElement('tbody');
+                            writeDescs(tbody, value, descriptionMapTime);
+                            table.append(tbody);
+                            statsView.append(table);
+                        }
+                    });
+                }
+            }
+        );
+    }
+
+    $(document).ready(function() {
+
+
+        $('#refresh').on('click', function() {
```

Comment:
any reason not to do an auto update ? it would increase UX, we don't have a static page to deal with :)

Reply:
I took inspiration from another stats page which didn't auto-update. That's the only reason, really, I think auto-updating is no problem.

How often do you think it should auto-update? (As in: once per x seconds/minutes?)

Reply:
somewhere between 10-30 seconds, probably on the lower half. whatever is good to read the values and understand them before they get flushed (e.g. copy values from the page before wiped again).

the firewall live log has a checkbox to stop auto-update... maybe that would satisfy all needs and the timeout can be as low as 10 seconds

Reply:
That sounded like a good plan so I implemented it. ^^